metrics.go

```diff
@@ -15,36 +14,43 @@ var (
 	msgSizeHistogram  prometheus.Histogram
 )
 
-const mb = 1024 * 1024
-
-func registerMetrics() {
-	requestsCounter = promauto.NewCounter(prometheus.CounterOpts{
+func init() {
+	requestsCounter = prometheus.NewCounter(prometheus.CounterOpts{
 		Namespace: "smtprelay",
 		Name:      "requests_count",
 		Help:      "count of message relay requests",
 	})
 
-	errorsCounter = promauto.NewCounterVec(prometheus.CounterOpts{
+	errorsCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
```

Comment:
whats the difference between using `promauto` here?

Reply:
`promauto` automatically registers using the default registry, but that's no good for testing, because the registry panics when you double-register the same metric.